chain/src/chain.rs

```diff
@@ -1217,9 +1193,9 @@ impl Chain {
 
 	/// Tip (head) of the header chain.
 	pub fn header_head(&self) -> Result<Tip, Error> {
-		self.store
-			.header_head()
-			.map_err(|e| ErrorKind::StoreErr(e, "chain header head".to_owned()).into())
+		let hash = self.header_pmmr.read().head_hash()?;
```

Comment:
I think this change breaks the TUI hashset validation percentage. Because that calls this function which now requires a read lock, which is not available as the validation itself has a write lock on header_pmmr.